index.d.ts

```diff
@@ -0,0 +1,250 @@
+type CurriedFunction2<T1, T2, R> = ((t1: T1, t2: T2) => R) & ((t1: T1, ...rest: Array<void>) => (t2: T2) => R);
+
+declare namespace flyd {
+  interface Morphism<T, V> {
+    (value: T): V;
+  }
+  interface Stream<T> {
+    (): T;
+    (value: T): Stream<T>;
+    (value: Promise<T> | PromiseLike<T>): Stream<T>;
+
+
+    /**
+     * @deprecated
+     */
+    map<V>(accessor: Morphism<T, V>): Stream<V>;
+    /**
+     * @deprecated
+     */
+    ap<A, B>(this: Stream<(value: A) => B>, stream: Stream<A>): Stream<B>;
+    /**
+     * @deprecated
+     */
+    of<V>(...values: V[]): Stream<V>;
+
+    pipe<V>(operator: (input: Stream<T>) => Stream<V>): Stream<V>;
+
+    ['fantasy-land/map']<V>(accessor: Morphism<T, V>): Stream<V>;
```

Comment:
Is there a reason for using `Morphism` instead of just `(a: A) => B`? I think the latter would be a bit more "mainstream"? Maybe even more idiomatic.

Reply:
No particular reason. Leftovers from other typings in my project.

I think accessor is the wrong name for the argument as well. How about `project`?